homeassistant/components/solaredge_local/sensor.py

```diff
@@ -112,6 +113,14 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         _LOGGER.error("Could not retrieve details from SolarEdge API")
         return
 
+    if status.inverters.primary.temperature.units.farenheit:
+        SENSOR_TYPES["inverter_temperature"] = [
```

Comment:
It's better to add this case as a nested dict under this key in `SENSOR_TYPES`. We don't want to modify `SENSOR_TYPES` since it's a constant defined at the module level.

Alternatively make a local copy of `SENSOR_TYPES` before modifying it and pass the copy to the entity.

Reply:
Like this?

```python

    global SENSORS
    SENSORS = deepcopy(SENSOR_TYPES)
    if status.inverters.primary.temperature.units.farenheit:
        SENSOR_TYPES["inverter_temperature"] = [
            "invertertemperature",
            "Inverter Temperature",
            TEMP_FAHRENHEIT,
            "mdi:thermometer",
        ]

    # Create solaredge data service which will retrieve and update the data.
    data = SolarEdgeData(hass, api)

    # Create a new sensor for each sensor type.
    entities = []
    for sensor_key in SENSORS:
        sensor = SolarEdgeSensor(platform_name, sensor_key, data)
        entities.append(sensor)

    add_entities(entities, True)
```

Reply:
We shouldn't use global. Pass the new sensors variable as an argument to the entity.